projects/dotcms-js/src/lib/core/core-web.service.ts

```diff
@@ -48,40 +50,53 @@ export const RULE_CONDITION_UPDATE_TYPE = 'RULE_CONDITION_UPDATE_TYPE';
 export const RULE_CONDITION_UPDATE_PARAMETER = 'RULE_CONDITION_UPDATE_PARAMETER';
 export const RULE_CONDITION_UPDATE_OPERATOR = 'RULE_CONDITION_UPDATE_OPERATOR';
 
+export interface RequestOptionsParams {
+    headers?: HttpHeaders;
+    reportProgress?: boolean;
+    params?: HttpParams;
+    responseType?: 'arraybuffer' | 'blob' | 'text' | 'json';
+    withCredentials?: boolean;
+}
+
 @Injectable()
 export class CoreWebService {
     private httpErrosSubjects: Subject<any>[] = [];
 
     constructor(
         private _apiRoot: ApiRoot,
-        private _http: Http,
         private loggerService: LoggerService,
         private browserUtil: BrowserUtil,
-        private router: Router
+        private router: Router,
+        private http: HttpClient
     ) {}
 
-    request(options: any): Observable<any> {
+    request(options: RequestOptionsArgs): Observable<any> {
         const request = this.getRequestOpts(options);
         const source = options.body;
 
-        return this._http.request(request).pipe(
-            map((resp: Response) => {
+        return this.http.request(request).pipe(
+            filter(<T>(event: HttpEvent<T>) => event.type === HttpEventType.Response),
+            map((resp: HttpResponse<any>) => {
                 // some endpoints have empty body.
                 try {
-                    return resp.json();
+                    return resp.body;
                 } catch (error) {
                     return resp;
                 }
             }),
             catchError(
-                (response: Response, _original: Observable<any>): Observable<any> => {
+                // tslint:disable-next-line:cyclomatic-complexity
+                (response: HttpResponse<any>, _original: Observable<any>): Observable<any> => {
                     if (response) {
                         this.handleHttpError(response);
                         if (
                             response.status === HttpCode.SERVER_ERROR ||
                             response.status === HttpCode.FORBIDDEN
                         ) {
-                            if (response.text() && response.text().indexOf('ECONNREFUSED') >= 0) {
+                            if (
+                                response.statusText &&
+                                response.statusText.indexOf('ECONNREFUSED') >= 0
```

Comment:
```suggestion
                                response?.statusText?.indexOf('ECONNREFUSED') >= 0
```

Reply:
![image](https://user-images.githubusercontent.com/37185433/94074459-85ddc400-fdb6-11ea-9785-122eae1b7153.png)

for some reason it does not allow it

Reply:
Weird, I'll check to see if I see something.

Reply:
not supported on current TS version, once upgraded to NG10, this can be done